src/ci/tested_versions_utils.py

```diff
@@ -1,38 +1,180 @@
+        return SemanticVersion(not bool(supported), int(major), int(minor), int(patch), suffix, comment)
+
+    (supported, version, comment) = re.search(_splitVersionFromCommentPattern, version).groups()
+    # The `supported` is either an empty string (supported) or the '!' string (not supported)
+    return NonSemanticVersion(not bool(supported), version, comment)
 
 
 @dataclass(frozen=True)
 class TestedVersions:
-    success: Tuple[str, ...]
-    failed: Tuple[str, ...]
+    versions: List[Union[SemanticVersion, NonSemanticVersion]]
 
     @staticmethod
     def _add_version_to_file(
-        directory: str, dependency_name: str, dependency_version: str, success: bool
+        directory: str, dependency_name: str, dependency_version: str, supported: bool
     ):
         dependency_file_path = TestedVersions.get_file_path(directory, dependency_name)
         TestedVersions.add_version_to_file(
-            dependency_file_path, dependency_version, success
+            dependency_file_path, dependency_version, supported
         )
 
     @staticmethod
-    def add_version_to_file(path: str, version: str, success: bool):
-        new_line = f"{'' if success else '!'}{version}\n"
-        print(f"Adding the following line to {path}: {new_line}")
-        with open(path, "a") as f:
-            f.write(new_line)
+    def add_version_to_file(path: str, version: str, supported: bool):
+        tested_versions = TestedVersions.from_file(path)
+
+        parsed_version = parse_version(('' if supported else '!') + version)
+
+        try:
+            previous_version = next(filter(lambda v: v.version == parsed_version.version, tested_versions.versions))
+        except StopIteration:
+            # This version does not appear in the file
+            previous_version = None
+            pass
+
+        tested_versions.versions.append(parsed_version)
+        if previous_version:
+            tested_versions.versions.remove(previous_version)
+
+            if parsed_version.supported == previous_version.supported:
+                print(f"Version '{parsed_version.version}' already marked as {'supported' if parsed_version.supported else 'not supported'} in {path}")
+            else:
+                print(f"Turning version '{parsed_version.version}' to {'supported' if parsed_version.supported else 'not supported'} in {path}")
+
+            if parsed_version.comment != previous_version.comment:
+                if parsed_version.comment:
+                    print(f"Updating comment for version '{parsed_version.version}' in {path}")
+                else:
+                    print(f"Removing comment for version '{parsed_version.version}' in {path}")
+        else:
+            print(f"Adding the {parsed_version.version} as {'supported' if parsed_version.supported else 'not supported'} to {path}")
+
+        with open(path, "w") as f:
+            for tested_version in sorted(tested_versions.versions):
```

Comment:
Please sort the versions files as part of this PR, otherwise the nightly will create bigger PRs which will be harder to understand.